3 course/OperatingSystem/http-proxy/http_proxy.c

```diff
@@ -0,0 +1,177 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <pthread.h>
+#include <netdb.h>
+
+#define MAX_BUFFER_SIZE 1024 * 4
+
+int setting_proxy_socket() {
+    struct sockaddr_in proxy_addr;
+    int proxy_socket;
+
+    proxy_socket = socket(AF_INET, SOCK_STREAM, 0);
+    if (proxy_socket < 0) {
+        perror("Ошибка при создании прокси-сокета");
+        exit(EXIT_FAILURE);
+    }
+
+    //настройка сокета для повторного использования порта
+    int opt = 1;
+    if (setsockopt(proxy_socket, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt))) {
+        perror("setsockopt");
+        exit(EXIT_FAILURE);
+    }
+
+    //настраиваем адрес и порт прокси-сокета
+    proxy_addr.sin_family = AF_INET;
+    proxy_addr.sin_addr.s_addr = INADDR_ANY;
+    proxy_addr.sin_port = htons(8080);
+
+    //привязываем прокси-сокет к адресу и порту
+    if (bind(proxy_socket, (struct sockaddr*)&proxy_addr, sizeof(proxy_addr)) < 0) {
+        perror("Ошибка при привязке прокси-сокета");
+        exit(EXIT_FAILURE);
+    }
+
+    //слушаем клиентиков
+    if (listen(proxy_socket, 5) < 0) {
```

Comment:
what does this magic number 5 mean and how does it influence on your program ?

Reply:
The backlog argument defines the maximum length to which the
       queue of pending connections for sockfd may grow.  If a
       connection request arrives when the queue is full, the client may
       receive an error with an indication of ECONNREFUSED or, if the
       underlying protocol supports retransmission, the request may be
       ignored so that a later attempt at connection succeeds.